File: chapitre-08/exemple-localtime.c
// ------------------------------------------------------------------
// exemple-localtime.c
// Fichier d'exemple du livre "Developpement Systeme sous Linux"
// (C) 2000-2019 - Christophe BLAESS <christophe@blaess.fr>
// https://www.blaess.fr/christophe/
// ------------------------------------------------------------------

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main (void)
{
	time_t      now;
	struct tm * tm;

	time(&now);
//	now += 180 * 86400;
	fprintf(stdout, "time() = %ld \n", now);
	tm = localtime(&now);
	fprintf(stdout, "localtime() = %02d/%02d/%02d - %02d:%02d:%02d %s\n",
		tm->tm_mday, tm->tm_mon + 1, tm->tm_year % 100,
		tm->tm_hour, tm->tm_min, tm->tm_sec,
		tm->tm_isdst>0 ? "Été" : tm->tm_isdst==0 ? "Normal" : "?");
		tm = gmtime (&now);
	fprintf(stdout, "gmtime()    = %02d/%02d/%02d - %02d:%02d:%02d %s\n",
		tm->tm_mday, tm->tm_mon + 1, tm->tm_year % 100,
		tm->tm_hour, tm->tm_min, tm->tm_sec,
		tm->tm_isdst>0 ? "Été" : tm->tm_isdst==0 ? "Normal" : "?");

	return EXIT_SUCCESS;
}

File: chapitre-08/exemple-localtime.py
#! /usr/bin/python
# ------------------------------------------------------------------
# exemple-localtime.py
# Fichier d'exemple du livre "Developpement Systeme sous Linux"
# (C) 2015-2019 - Christophe BLAESS <christophe@blaess.fr>
# https://www.blaess.fr/christophe/
# ------------------------------------------------------------------

from __future__ import print_function
import time

t = time.time()

print("time.time()      =", t)

tm = time.localtime(t)
if (tm.tm_isdst > 0):
	dst = "Ete"
elif (tm.tm_isdst == 0):
	dst = "Normal"
else:
	dst = "?"
print("time.localtime() = {:02}/{:02}/{:02} {:02}:{:02}:{:02} {}".format(
       tm.tm_mday, tm.tm_mon, tm.tm_year % 100,
       tm.tm_hour, tm.tm_min, tm.tm_sec, dst))

tm = time.gmtime(t)
if (tm.tm_isdst > 0):
	dst = "Ete"
elif (tm.tm_isdst == 0):
	dst = "Normal"
else:
	dst = "?"
print("time.gmtime()    = {:02}/{:02}/{:02} {:02}:{:02}:{:02} {}".format(
       tm.tm_mday, tm.tm_mon, tm.tm_year % 100,
       tm.tm_hour, tm.tm_min, tm.tm_sec, dst))
